query_executor.go

```diff
@@ -152,6 +152,9 @@ func (q *queryExecutor) run(qry ExecutableQuery, specWG *sync.WaitGroup, results
 			case RetryNextHost:
 				// retry on the next host
 				selectedHost = hostIter()
+				if selectedHost == nil {
+					results <- queryResponse{iter: iter}
+				}
```

Comment:
right, I thought it's covered but the main loop....so to avoid confusion, let's return after this. Alternatively, we can move the channel loading to after the loop, and check if the iter pointer is not nil, then push it to the results channel. I actually have no preference here on one way or the other.